plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/ElkUtil.java

```diff
@@ -396,6 +398,75 @@ public static void applyConfiguredNodeScaling(final ElkNode node) {
             }
         }
     }
+    
+    /**
+     * Computes the area occupied by this node's layout and stores the values in {@link CoreOptions#CHILD_AREA_WIDTH} 
+     * and {@link CoreOptions#CHILD_AREA_HEIGHT}.
+     * @param node the node whose child area should be computed
+     */
+    public static void computeChildAreaDimensions(final ElkNode node) {
+
+        double minX = Double.POSITIVE_INFINITY;
+        double minY = Double.POSITIVE_INFINITY;
+        double maxX = 0.0;
+        double maxY = 0.0;
+        
+        // iterate over all nodes and labels and get their coordinate bounds
+        Iterable<ElkLabel> edgeLabels = new ArrayList<>();
+        for (ElkEdge edge : node.getContainedEdges()) {
+            edgeLabels = Iterables.concat(edgeLabels, edge.getLabels());
+        }
+        for (ElkShape shape : Iterables.concat(node.getLabels(), node.getChildren(), edgeLabels)) {
+
+            ElkMargin margins = shape.getProperty(CoreOptions.MARGINS);
```

Comment:
In which case can't you just calculate the bounds? Why do you need the maximum/minimum checks?